### PR TITLE
Add a bash function to prompt for secret value and add to keystore

### PR DESCRIPTION
**Pull request by shubha-rajan**

### CODE REVIEW

File: cloud/azure/bin/lib/key_vault.sh [hunk 1]
```diff
@@ -57,6 +57,21 @@ function key_vault::add_secret() {
...
+#   1: The name of the key vault
+#   2: The name of the secret (used to identify it e.g. "postgres-password")
+#######################################
+function key_vault::add_secret_from_input() {
+  local SECRET
+  echo "Please enter the value for ${2}: "
+  read -s $SECRET && unset REPLY
+  key_vault::add_secret "${1}" "${2}" "${SECRET}" && \
+    echo "Stored secret value for ${2} in key vault ${1}"
+
```
Comment (bion) [inline cloud/azure/bin/lib/key_vault.sh:72]: nit: remove extra newline

File: cloud/azure/bin/lib/key_vault.sh [hunk 2]
```diff
@@ -57,6 +57,21 @@ function key_vault::add_secret() {
...
+# Arguments:
+#   1: The name of the key vault
+#   2: The name of the secret (used to identify it e.g. "postgres-password")
+#######################################
+function key_vault::add_secret_from_input() {
+  local SECRET
+  echo "Please enter the value for ${2}: "
+  read -s $SECRET && unset REPLY
+  key_vault::add_secret "${1}" "${2}" "${SECRET}" && \
+    echo "Stored secret value for ${2} in key vault ${1}"
```
Comment (bion) [inline cloud/azure/bin/lib/key_vault.sh:71]: suggestion (blocking): we `set -e` in all of our scripts, so these commands can be run sequentially with the same behavior
```suggestion
  read -s $SECRET
  unset REPLY
  key_vault::add_secret "${1}" "${2}" "${SECRET}"  
  echo "Stored secret value for ${2} in key vault ${1}"
```

File: cloud/azure/bin/lib/key_vault.sh [hunk 3]
```diff
@@ -57,6 +57,21 @@ function key_vault::add_secret() {
...
+#######################################
+# Add a secret to the key vault
+# Arguments:
+#   1: The name of the key vault
+#   2: The name of the secret (used to identify it e.g. "postgres-password")
+#######################################
+function key_vault::add_secret_from_input() {
+  local SECRET
+  echo "Please enter the value for ${2}: "
+  read -s $SECRET && unset REPLY
```
Comment (bion) [inline cloud/azure/bin/lib/key_vault.sh:69]: question (blocking): why `unset REPLY`?

**Review by bion — CHANGES_REQUESTED** [review]

File: cloud/azure/bin/lib/key_vault.sh (hunk 3 above)
Comment (shubha-rajan) [reply cloud/azure/bin/lib/key_vault.sh:69]: I was thinking we didn't want secret values hanging out in environment variables

Comment (bion) [reply cloud/azure/bin/lib/key_vault.sh]: shubha explained to me in chat, this is a best practice

**Review by bion — APPROVED** [review]

**Review by goldblatt — APPROVED** [review]

**Pull request merged**